app/src/main/java/org/jellyfin/mobile/utils/LocaleUtils.kt

```diff
@@ -0,0 +1,53 @@
+package org.jellyfin.mobile.utils
+
+import android.app.Activity
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.launch
+import org.jellyfin.mobile.MainActivity
+import timber.log.Timber
+import java.util.*
+
+@Suppress("DEPRECATION")
+fun MainActivity.initLocale() {
+    webView.evaluateJavascript("window.localStorage.getItem('${apiClient.currentUserId}-language')") { result ->
+        if (!setLocale(result.unescapeJson(), "userSettings")) {
+            lifecycleScope.launch {
+                if (!setLocale(apiClient.getServerConfiguration()?.uiCulture,"serverSettings")) {
+                    val locale = resources.configuration.locale
+                    setLocale("${locale.language}-${locale.country}", "androidSettings")
+                }
+            }
+        }
+    }
+}
+
+@Suppress("DEPRECATION")
+private fun Activity.setLocale(localeString: String?, source: String): Boolean {
+    var language: String? = null
+    var country: String? = null
+
+    if (localeString.isNullOrEmpty() || localeString == "null") {
+        return false
+    } else {
+        if (localeString.contains('-')) {
+            val cultureSplit = localeString.split('-')
+            if (cultureSplit.size == 2) {
+                language = cultureSplit[0]
+                country = cultureSplit[1]
+            }
+        } else {
+            language = localeString
+            country = ""
+        }
+        Timber.d("setLocale [source: $source, language: '${language}', country: '${country}']")
+        val configuration = resources.configuration
+        val locale = Locale(
+            language ?: configuration.locale.language,
+            country ?: configuration.locale.country
+        )
+        Locale.setDefault(locale)
+        configuration.setLocale(locale)
+        resources.updateConfiguration(configuration, resources.displayMetrics)
```

Comment:
Could you only update the configuration if there's actually a locale set, and maybe even check if the locales differ?